Use pylint config parser for pyproject.toml and setup.cfg support

kevinoid opened the pull request:
Pylint 2.5.0 and later support reading configuration from `pyproject.toml` or `setup.cfg` if `pylintrc` and `.pylintrc` do not exist.  (PyCQA/pylint#3169)  In order to support these configuration files, this PR changes pylint_runner to use `pylint.lint.pylinter.PyLinter#read_config_file()` instead of `configparser.ConfigParser#read()` to read the pylint configuration.

It also adds a check to avoid `configparser.NoOptionError` if `ignore` is not configured and to strip whitespace from `ignore` values as pylint does.  If you would prefer these to be moved to a separate PR, let me know.

Thanks,
Kevin

kevinoid committed:
use pylint configuration parser

Pylint 2.5.0 and later support reading configuration from
`pyproject.toml` or `setup.cfg` if `pylintrc` and `.pylintrc` do not
exist.  (PyCQA/pylint#3169)  In order to support these configuration
files, use `pylint.lint.pylinter.PyLinter#read_config_file()` instead of
`configparser.ConfigParser#read()` to read the pylint configuration.

Signed-off-by: Kevin Locke <kevin@kevinlocke.name>

kevinoid committed:
avoid NoOptionError if ignore not configured

If the `ignore` option is not configured, pylint_runner dies with:

    configparser.NoOptionError: No option 'ignore' in section: 'MASTER'

Avoid this by checking that the option is present using `#has_option()`
before `#get()`.  (Note: It is not necessary to check that the `MASTER`
section exists, since `#has_option()` returns `False` in that case.)

Signed-off-by: Kevin Locke <kevin@kevinlocke.name>

kevinoid committed:
parse ignore as pylint does

`pylint.lint.pylinter.PyLinter#set_option()` ultimately calls
`pylint.utils.utils._splitstrip()` to parse options of type `csv`, such
as `ignore`.  In addition to removing empty values, `_splitstrip()`
strips whitespace from values.  Update `ignore` parsing to behave the
same as pylint.

Note: Since `_splitstrip()` is private and has a simple implementation,
it is inlined here.

Signed-off-by: Kevin Locke <kevin@kevinlocke.name>

kevinoid commented:
For reference, the test failures don't appear to be related to this PR (since it doesn't change `setup.py` or `tests/test_runner.py` which are causing the pylint check to fail).  I'd be happy to rebase this PR once those issues are fixed.

MasterOdin commented:
Sorry about that @kevinoid. Master has been fixed and tests are passing once again.

kevinoid committed:
avoid overwriting Runner.DEFAULT_ARGS

Copy `Runner.DEFAULT_ARGS` to `self.args` so that if/when `self.args` is
modified it will not affect `Runner.DEFAULT_ARGS`.

Fix the tests as a result of the change:
1. Increase `max-line-length` in `tests/tests/good_rc_file` so that when
   it is used in `test_rcparser_success` the lint succeeds.
2. Fix `expected` in `test_passthrough_args` to not expect args to have
   been modified by previous tests.

Signed-off-by: Kevin Locke <kevin@kevinlocke.name>

kevinoid commented:
No worries.  Thanks for the quick fix @MasterOdin!  It should pass the CI tests now.

I added an additional commit to avoid modifying `Runner.DEFAULT_ARGS` (4cb3cee051de5cecb1821494cace319356335693) which was causing the changes to fail the test unnecessarily.  If you'd like to consider that as a separate PR first, let me know.
